enzy_htp/_interface/rosetta_interface.py

```diff
@@ -1499,4 +1499,129 @@ def build_cartesian_ddg_engine(
         )
 
     # endregion
- 
+
+    def add_missing_residues(self, 
+            stru: Structure,
+            missing_residues:List[SeqRes],
+            work_dir:str=None,
+            **kwargs
+        ) -> None: 
+        """Uses the LoopRemodel mover to add missing loop residues to a Structure(). All work is done 
+        to the Structure() in place.
+
+        Args:
+            stru: The Structure() with missing loop Residue()'s.
+            missing_residues: The List[SeqRes] of missing Residues.
+            work_dir: The directory where work will take place. Optional.
+
+        Returns:
+            Nothing.
+        """
+        
+        fs.safe_mkdir( work_dir )
+
+        translate_structure( stru, end_naming='rosetta' )
+        
+        nc_chains = list()
+        aa_chains = list()
+
+        for chain in stru.chains:
+            if not len(chain.residues):
+                continue
+            
+            if chain.residues[0].is_canonical():
+                aa_chains.append( chain )
+            else:
+                nc_chains.append( chain )
```

Comment:
Be great if we can put this in Structure

Reply:
It could be a method that returns the AA sequence of each chain.

Reply:
Check out Structure().sequence. We can make a Structure().seqres_sequence

Reply:
sure!

Reply:
To clean this up, I have implemented the `aa_chains()` and `non_aa_chains()` methods. These are separate since the non-amino acid chains need to be copied to be added back into the structure at the end. Your point about sequence is well taken, and I have added that as well as the `seqeres_sequence()` method.